### PR TITLE
Feature/player weapons

### PR DESCRIPTION
**Pull request by OttoRocket-LVP**
Add the player_weapons feature to make it able to assign weapons to players in JS

### CODE REVIEW

File: javascript/features/player_weapons/player_weapons_manager.test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    beforeEach(() => {
+        playerWeapons = server.featureManager.loadFeature('player_weapons');
+
+        manager = playerWeapons.manager_;
+    });
+
+    it('should invoke the pawn command upon trying to give a weapon.', assert => {
+        let counter = 0;
+
+        var mockPawnInvoke = MockPawnInvoke.getInstance();
```
Comment (RussellLVP) [inline javascript/features/player_weapons/player_weapons_manager.test.js]: nit: here and below, `var` --> `const`

**Review by RussellLVP — COMMENTED** [review]
What is your plan with this supplement? Wouldn't it be easier to just have these methods on `Player` directly, as they don't rely on any feature-specific behaviour? (A feature could provide commands.)

**Comment by OttoRocket-LVP** [comment]
The reason to put this into a supplement is:

We currently track what weapons are in what slot for the player, have them able to get spawnweapons (and thus store what spawnies they have) and probably more. I think it would be best to keep that logic away from the player.

**Comment by RussellLVP** [comment]
I think that keeping track of actual player data, e.g. weapons they're carrying, is perfectly fine for the Player class. It inherently is something that's part of the Player's state, the server knows, just isn't exposing to scripts for whatever reason. We do the same with e.g. `Vehicle.paintjob`,

Regarding spawn weapons, yeah, that absolutely shouldn't be in the `Player` class. In that case, though, I'd ask the question why that couldn't be a _spawn weapons_ feature, which keeps track of spawn weapons in a map -- why does _all_ other code need to be able to be able to access that state? Wouldn't a dependency on whatever Feature provides the spawn weapon data be sufficient?

**Comment by OttoRocket-LVP** [comment]
I don't agree nor disagree;

However, we don't need to fix an issue we _might_ have later as we don't know what the future brings.

**Pull request merged**